app/src/main/java/org/simple/clinic/enterotp/BruteForceOtpEntryProtection.kt

```diff
@@ -26,4 +40,69 @@ class BruteForceOtpEntryProtection @Inject constructor(
       otpState.loginAttemptFailed()
     }
   }
+
+  fun protectedStateChanges(): Observable<ProtectedState> {
+    val bruteForceProtectionResets = Observables
+        .combineLatest(config, otpStatePreference.asObservable()) { config, state -> config.blockDuration to state.limitReachedAt }
+        .switchMap { (blockDuration, blockedAt) -> signalBruteForceTimerReset(blockedAt, blockDuration) }
+        .flatMapCompletable { resetFailedAttempts() }
+        .toObservable<Any>()
+
+    return Observables
+        .combineLatest(
+            config,
+            otpStatePreference.asObservable(),
+            bruteForceProtectionResets.startWith(Any())
+        )
+        .map { (config, state) ->
+          generateProtectedState(
+              blockedAt = state.limitReachedAt,
+              attemptsMade = state.failedLoginOtpAttempt,
+              maxAllowedFailedAttempts = config.limitOfFailedAttempts,
+              blockAttemptsFor = config.blockDuration
+          )
+        }
+        .distinctUntilChanged()
```

Comment:
```suggestion
val bruteForceProtectionResets = otpStatePreference
    .asObservable()
    .switchMap { state -> signalBruteForceTimerReset(state.limitReachedAt, config.blockDuration) }
    .doOnNext { resetFailedAttempts() }
return Observables
    .combineLatest(
        otpStatePreference.asObservable(),
        bruteForceProtectionResets
    )
    .map { (state) ->
      generateProtectedState(
          blockedAt = state.limitReachedAt,
          attemptsMade = state.failedLoginOtpAttempt,
          maxAllowedFailedAttempts = config.limitOfFailedAttempts,
          blockAttemptsFor = config.blockDuration
      )
    }
    .distinctUntilChanged()
```

Overall, something like this perhaps? WDYT?